modules/fluid_properties/src/userobjects/BrineFluidProperties.C

```diff
@@ -0,0 +1,331 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "BrineFluidProperties.h"
+
+template<>
+InputParameters validParams<BrineFluidProperties>()
+{
+  InputParameters params = validParams<MultiComponentFluidPropertiesPT>();
+  params.addClassDescription("Fluid properties for brine");
+  return params;
+}
+
+BrineFluidProperties::BrineFluidProperties(const InputParameters & parameters) :
+    MultiComponentFluidPropertiesPT(parameters),
+    _Mnacl(58.443e-3)
+{
+  // Water97FluidProperties UserObject for water (needed to access pSat)
+  std::string water97_name = "water97";
+  {
+    std::string class_name = "Water97FluidProperties";
+    InputParameters params = _app.getFactory().getValidParams(class_name);
+    _fe_problem.addUserObject(class_name, water97_name, params);
+  }
+  _water97_fp = &_fe_problem.getUserObject<Water97FluidProperties>(water97_name);
+
+  // SinglePhaseFluidPropertiesPT UserObject for water to provide to getComponent
+  std::string water_name = "water";
+  {
+    std::string class_name = "Water97FluidProperties";
+    InputParameters params = _app.getFactory().getValidParams(class_name);
+    _fe_problem.addUserObject(class_name, water_name, params);
+  }
+  _water_fp = &_fe_problem.getUserObject<SinglePhaseFluidPropertiesPT>(water_name);
+
+  // SinglePhaseFluidPropertiesPT UserObject for NaCl to provide to getComponent
+  std::string halite_name = "halite";
+  {
+    std::string class_name = "NaClFluidProperties";
+    InputParameters params = _app.getFactory().getValidParams(class_name);
+    _fe_problem.addUserObject(class_name, halite_name, params);
+  }
+  _halite_fp = &_fe_problem.getUserObject<SinglePhaseFluidPropertiesPT>(halite_name);
```

Comment:
For generating the names of these user objects, you want to do something like `name() + ":" + internal_object_name`. The reason is that names of MOOSE objects have to be unique and this will prevent some name collision. "water", "water97" and "halite" are too likely to show up in an input file and than it would fail running.